diff --git a/agent/auto-config/server_addr.go b/agent/auto-config/server_addr.go
index 6bca15d..0c9767c 100644
--- a/agent/auto-config/server_addr.go
+++ b/agent/auto-config/server_addr.go
@@ -12,6 +12,7 @@ import (
 	"github.com/hashicorp/consul/lib"
 	"github.com/hashicorp/go-discover"
 	discoverk8s "github.com/hashicorp/go-discover/provider/k8s"
+	discoverdockerswarm "github.com/socheatsok78/go-discover-dockerswarm"
 
 	"github.com/hashicorp/go-hclog"
 )
@@ -22,6 +23,7 @@ func (ac *AutoConfig) discoverServers(servers []string) ([]string, error) {
 		providers[k] = v
 	}
 	providers["k8s"] = &discoverk8s.Provider{}
+	providers["dockerswarm"] = &discoverdockerswarm.Provider{}
 
 	disco, err := discover.New(
 		discover.WithUserAgent(lib.UserAgent()),
